api/godog/json/matcher.go

```diff
@@ -55,270 +52,76 @@ func Match(json []byte, path string, pattern string) error {
 		return errors.Wrap(err, "Failed to compile regexp")
 	}
 
-	t, err := resolve(json, path)
-	if err != nil {
-		return err
-	}
+	path = revertLegacySyntax(path)
 
-	value, ok := t.value.(string)
-	if !ok {
-		return errors.Errorf("Match error: Expected a scalar got '%T' JSON: %s", t.value, string(json))
-	}
+	result := gjson.Get(string(json), path)
 
-	if !re.MatchString(value) {
-		return errors.Errorf("Match error: Values mismatch, pattern: '%s' value: '%s' JSON: %s", pattern, value, string(json))
+	if !re.MatchString(result.String()) {
+		return errors.Errorf("Match error: Values mismatch, pattern: '%s' value: '%s' JSON: %s", pattern, result.String(), string(json))
 	}
 
 	return nil
 }
 
-func ArrayLen(json []byte, path string, length int) error {
-	t, err := resolve(json, path)
-	if err != nil {
-		return err
-	}
-
-	value, ok := t.value.([]term)
-	if !ok {
-		return errors.Errorf("Match error: Expected an array got '%T' JSON: %s", t.value, string(json))
-	}
-
-	if len(value) != length {
-		return errors.Errorf("Match error: Expected an array of length: %d got: %d JSON: %s", length, len(value), string(json))
-	}
-	return nil
-}
-
-func Read(json []byte, path string) (result string, err error) {
-	t, err := resolve(json, path)
-	if err != nil {
-		return "", err
-	}
-
-	value, ok := t.value.(string)
-	if !ok {
-		return "", errors.Errorf("Match error: Expected a scalar got '%T' JSON: %s", t.value, string(json))
-	}
-
-	return value, nil
-}
-
-func ReadStringArr(json []byte, path string) (result []string, err error) {
-	t, err := resolve(json, path)
-	if err != nil {
-		return []string{}, err
-	}
+func revertLegacySyntax(path string) string {
+	// Remove brackets earlier they were used for array indexing
+	bracketRegexp := regexp.MustCompile(`\.?\[(\d+)\]`)
```

Comment:
Why `\.?` in the beginning of the regexp? An optional character before the brackets?

Reply:
Added a test case to show where the initial dot is needed.

in case of path = .[0]

Reply:
I see, `\.` matches an actual dot :+1: